src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -377,3 +376,47 @@ def _reject_all_pending_requests(self):
                                   f"on area {self.device.name} with arguments {req.arguments}."
                                   f"Market cycle already finished."})
         self.pending_requests = []
+
+    def _set_power_forecast(self, payload):
+        transaction_id = self._get_transaction_id(payload)
+        power_forecast_response_channel = \
+            f'{self.channel_prefix}/response/set_power_forecast'
+        if not check_for_connected_and_reply(self.redis, power_forecast_response_channel,
+                                             self.connected):
+            return
+        try:
+            arguments = json.loads(payload["data"])
+            assert set(arguments.keys()) == {'power_forecast', 'transaction_id'}
+        except Exception as e:
+            logging.error(
+                f"Incorrect _set_power_forecast request. "
+                f"Payload {payload}. Exception {str(e)}.")
+            self.redis.publish_json(
+                power_forecast_response_channel,
+                {"command": "set_power_forecast",
+                 "error": "Incorrect _set_power_forecast request. "
+                          "Available parameters: (power_forecast).",
+                 "transaction_id": transaction_id})
+        else:
+            self.pending_requests.append(
+                IncomingRequest("set_power_forecast", arguments,
+                                power_forecast_response_channel))
+
+    def _set_power_forecast_impl(self, arguments, response_channel):
+        try:
+            assert arguments["power_forecast"] >= 0.0
+            self.power_forecast_buffer_W = arguments["power_forecast"]
```

Comment:
Might be an overkill but I think that in the long run there needs to be an extra validation here, that will check the power forecast against the nameplate characteristics of the load and PV (essentially the max PV/Load power)

Reply:
I agree in general, but then we need to add this as an input argument for the new Strategy which will also include a UI change. So i suggest to do it later. Shall I add a ticket?

Reply:
Please do so yes, but as I mentioned above this is for the long run, not urgent for the hackathon. Thanks!

Reply:
Done https://gridsingularity.atlassian.net/browse/D3ASIM-3010